daemon/oci_linux.go

```diff
@@ -98,6 +100,31 @@ func setDevices(s *specs.Spec, c *container.Container) error {
 		for _, d := range hostDevices {
 			devs = append(devs, oci.Device(d))
 		}
+		for _, deviceMapping := range c.HostConfig.Devices {
+			if deviceMapping.PathOnHost == deviceMapping.PathInContainer {
+				logrus.WithField("container", c.ID).Warnf("Specifying a device %s is redundant for privileged mode", deviceMapping.PathOnHost)
+				if deviceMapping.CgroupPermissions != "rwm" {
+					logrus.WithField("container", c.ID).Warnf("Specifying permission %s to device %s is redundant for privileged mode", deviceMapping.CgroupPermissions, deviceMapping.PathOnHost)
```

Comment:
I'd prefer this warning to look more like a warning; it's not really "redundant", but will be ignored (so you'd end up with different permissions than you configured);

Something like:

```
running in privileged mode: custom %s permissions for device %s are ignored
```

or

```
custom %s permissions for device %s are ignored in privileged mode
```

Reply:
Done